cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/roller/CommandPlannerImpl.java

```diff
@@ -0,0 +1,30 @@
+package io.strimzi.operator.cluster.operator.resource.roller;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Optional;
+
+public class CommandPlannerImpl implements CommandPlanner {
+    @Override
+    public RollCommand planNextCommand(Collection<Context> contexts) {
+        Optional<Context> nextToActOn = contexts.stream().min(contextPriority());
+        if(nextToActOn.isEmpty()){
+            throw new IllegalStateException("No context selected to act on");
+        }
+        Context nextContext = nextToActOn.get();
+        var state = nextContext.state();
+        if (state.isHealthy()) {
+            // Even the worst state is healthy => We're done
+            return new RollCommand(RollCommand.Command.END_ROLL, null);
+        } else if (state == State.STALE_BLOCKED) {
+            return new RollCommand(RollCommand.Command.DELAY, null);
+        } else {
+            return new RollCommand(RollCommand.Command.PROCESS_CONTEXT, nextContext);
+        }
+    }
+
+    private static Comparator<Context> contextPriority() {
```

Comment:
to ensure we pick controllers last we may need to know more than just the classified state. Maybe the Context should keep a reference to the latest observation so that it can also offer more information about the pod, like `isController`.